pynamodb/models.py

```diff
@@ -324,15 +335,42 @@ def __repr__(self):
                 msg = "{}<{}>".format(self.Meta.table_name, serialized.get(HASH))
             return six.u(msg)
 
-    def delete(self, condition=None):
+    @classmethod
+    def condition_check(cls, hash_key, in_transaction, condition, range_key=None):
+        hash_key, range_key = cls._serialize_keys(hash_key, range_key)
+        operation_kwargs = cls._get_connection().get_operation_kwargs_for_condition_check(
+            hash_key=hash_key,
+            range_key=range_key,
+            condition=condition,
+        )
+        in_transaction.add_condition_check_item(
+            model_cls=cls,
+            hash_key=hash_key,
+            range_key=range_key,
+            operation_kwargs=operation_kwargs
+        )
+
+    def delete(self, condition=None, in_transaction=None):
```

Comment:
`get`s and `put`s are methods of the transaction, but `delete`s still use the `in_transaction`-parameter API?

Reply:
yeah, we just changed it for `get` because we're returning something, but I didn't think we'd agreed to make all of them methods on the transaction? cc @garrettheel

Reply:
I think it was @jpinner-lyft 's strong opinion that methods should preferably have two separate return types based on a parameter

Reply:
o, i meant that we're returning something _new_. Like, `Model.get` normally returns an instance of that model, populated with the data returned from the request. But, because we don't have that data until after a transaction completes, we made the decision to return something *completely* different (a `ModelFuture`) to make it clearer to the user that something has to happen before they're able to access the complete model.

The same isn't necessarily true for any of the other operations (except for maybe the `Model.update` because it normally ends with populating the model with the updated data)

Reply:
```python
if in_transaction is not None:
    ...
    return {}  # empty dict
return self._get_connection().delete_item(*args, **kwargs)  # actual API response
```

so, yeah, the type technically is the same, but the nature is arguably not the same (and arguably that empty dict should've been a `None`)

Reply:
For sake of argument, let's say this function would be typed to return `Optional[dict]` - that'd be suboptimal, since its parameter `is_transaction` controls which one it'd be, so we again come to:
```python
@overload
def delete(..., in_transaction: Transaction) -> None: ...
@overload
def delete(..., in_transaction: None) -> dict: ...
```

Reply:
im fine with conditionally returning `None` :0